app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/components/charge_item.scala.html

```diff
@@ -16,16 +16,50 @@
 
 @import play.api.i18n.Messages
 @import play.twirl.api.Html
-
+@import java.time.LocalDate
 @import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.returns.routes
 @import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.finance.Charge
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.finance.Payment
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.finance.ClearingReason._
 @import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.TimeUtils
 @import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.finance.MoneyUtils
 
 @this()
 
 @(charge: Charge)(implicit messages: Messages)
 
+    @makePaymentDetailsHtml(paymentItemSubKey : String, paymentItemValue: String, amount:BigDecimal, clearingDate: LocalDate) = {
+        <li>
+            <strong class="bold">
+                @{MoneyUtils.formatAmountOfMoneyWithPoundSign(amount)}
+            </strong>
+            @{messages(s"viewReturn.charge.$paymentItemSubKey.$paymentItemValue")}
+            @{messages("generic.on")}
+            @{TimeUtils.govShortDisplayFormat(clearingDate)}
+        </li>
+    }
+
+    @displayPaymentDetails(payment : Payment) = @{
+        (payment.method, payment.clearingReason) match {
+            case (Some(paymentMethod), _) =>
+                    makePaymentDetailsHtml("paymentMethod", paymentMethod.toString, payment.amount.inPounds(), payment.clearingDate)
+            case (None, Some(clearingReason)) => clearingReason match {
+                case IncomingPayment | OutgoingPayment | WriteOff | Reversal | MassWriteOff | AutomaticClearing | SomeOtherClearingReason =>
+                    makePaymentDetailsHtml("clearingReason", clearingReason.toString, payment.amount.inPounds(), payment.clearingDate)
+                case InvalidClearingReason if (payment.method.isDefined) =>
+                    makePaymentDetailsHtml("paymentMethod", payment.method.fold("")(_.toString), payment.amount.inPounds(), payment.clearingDate)
+                case InvalidClearingReason if (!payment.method.isDefined) =>
+                    <li>
+                        {messages(s"viewReturn.charge.display-error")}
```

Comment:
the intention here was to just display the technical error page if there was no clearing reason or payment method. We don't have content for this case.

Reply:
It's impossible not to get both those data points. But to complete the case analysis something had to be put there.

Reply:
if we had an `Either[ClearingReason,PaymentMethod]` we could avoid handling cases which we don't expect - but that does come at the cost of backwards compatibility which wouldn't result in permanent failures in this case

Reply:
let me know you what you decide.

Reply:
if we had them both as `Option` then to do what was intended we'd have to error out somewhere in the FE. If it was an `Either` then it prevents the FE from having to do anything special. However, if we ever did want to not block the user from seeing the page if we couldn't get a payment method and to just display some sort of message which is what you've done here, then having options now would make that easier in the future. It's not unfeasible that we wouldn't want to do that cos that piece of text doesn't prevent them from doing other things, so let's leave them as `Option` for now and maybe revisit this case in the future to see if we want to display some sort of message like you've done here.